doc/source/deploy-guide/overview.rst

```diff
@@ -0,0 +1,73 @@
+Overview
+========
+
+This guide refers to the following type of hosts:
+
+* A `localhost`, which runs shell scripts and ansible playbooks. This can
+  be your CI node, or your development laptop.
+* A `deployer`, which is your point of access to your kubernetes
+  cluster. The `deployer` can be the same as the `localhost`, but it
+  is not a requirement.
+* A series of `CaaSP` nodes: `workers`, `admin`, `master`.
+* A series of `SES` nodes.
+
+The following diagram shows the general workflow of a deployment from scratch:
+
+.. blockdiag::
+
+   blockdiag {
+
+     localhost [label="Prepare localhost"]
+     ses [label="Deploy SES\n(optional)"]
+     caasp [label="Deploy CaaSP\n(optional)"]
+     deployer [label="Deploy deployer\n(optional)"]
+     enroll_caasp [label="Enroll CaaSP\n(optional)"]
+     setup_caasp_workers [label="Setup CaaSP\nfor OpenStack"]
+     patch_upstream [label="Apply patches\nfrom upstream\n(for developers)"]
+     build_images [label="Build docker images\n(for developers)"]
+     deploy [label="Deploy Airship or\nOpenStack-Helm"]
+     configure_deployment [label="Configure deployment"]
+
+
+     localhost -> ses;
+
+     group {
+       color = "#EEEEEE"
+       label = "Setup hosts"
+       ses -> caasp;
+       caasp -> deployer [folded];
+       deployer -> enroll_caasp;
+     }
+     enroll_caasp -> configure_deployment [folded];
+     localhost -> configure_deployment[folded];
+
+     configure_deployment -> setup_caasp_workers;
+
+     group {
+       color = "#EEEEEE"
+       label = "Setup openstack/Setup airship"
+       setup_caasp_workers -> deploy, patch_upstream [folded];
+       patch_upstream -> build_images;
+       build_images -> deploy;
+     }
+   }
+
+Installation requirements
+-------------------------
+
+Your environment should be setup with the following:
+
+* `localhost` can run any OS. Please check its software requirements on the
+  next page.
+
+* `deployer` must run openSUSE Leap 15 or SLE15. Those must have all the
+  deployment tools available. If you are not coming with your own node for
```

Comment:
Should we list required deployment tools here? Or just mention that ansible tasks are there for the preparation of a deployer?

Reply:
Deployer should have repos correctly configured, IIRC, should people bring their own deployers. It would be wise to document the requirements, which are currently git + all the caasp tooling requirements (ruby/jq/...)

Reply:
> Deployer should have repos correctly configured

Well, that's what I mean, sum up somewhere what does it take to correctly configure the deployer. We might leave specific for manual setup for later (but we should have a TODO somewhere) and point to the fact that ansible currently does it when user lets it deploy the deployer